project_issue_task_sync/models/project_issue.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class ProjectIssue(models.Model):
+    _inherit = 'project.issue'
+
+
+    def get_changed_vals(self, task):
+        vals = {}
+        if len(task.issue_ids) ==1:
+            if task.stage_id != self.stage_id:
+                vals['stage_id'] = self.stage_id.id
+            if task.user_id != self.user_id:
+                vals['user_id'] = self.user_id.id
+        return vals
+
+
+    def set_issue_vals(self):
+        for this in self:
+            task = this.sudo().task_id
+            """
+            if the create/write is called by a task creation skip all 
+            syncing functions and do not try to create a task
+            if it is not called vy a task creation 
+            and the project is "sync enabled" create the associated 
+            task.
+            """
+            if (this.project_id.sync_tasks_issues and not 
+                   self.env.context.get('sync_operation', False)):
```

Comment:
no, it has been tested. this means: "this write is being done not by the user but is the result of a sync operation already"   there fore  don't do the sync operation again. it's a structure where issue and task call sync eachother. on write they are written and sync the other model with sync_operation =True so that the other model's write won't sync back again. works perfect.